### PR TITLE
Replace aioredis with redis 4.3 or higher

### PR DESCRIPTION
**Pull request by achimnol**
## What do these changes do?

It replaces aioredis v2 dependency with redis v4.3 or higher,
in favor of official aioredis inclusion in the redis-py project since v4.2.

## Are there changes in behavior for the user?

Redis storage users should update their aioredis v2 dependency to redis v4.3.

## Related issue number

This is a standalone PR.

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` (e.g. `588.bugfix`)
  * if you don't have an `issue_id` change it to the pr id after creating the PR
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: `Fix issue with non-ascii contents in doctest text files.`

### CODE REVIEW

**Comment by achimnol** [comment]
Reported spurious type check failures as python/typeshed#7837.

File: aiohttp_session/redis_storage.py [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-from distutils.version import StrictVersion
 from typing import Any, Callable, Optional
 
 from aiohttp import web
 
 from . import AbstractStorage, Session
 
 try:
-    import aioredis
+    import redis
```
Comment (Dreamsorcerer) [inline aiohttp_session/redis_storage.py:10]: I'd consider just doing `from redis import __version__ as redis_version`. Which will also avoid the lgtm warning.

File: aiohttp_session/redis_storage.py [hunk 2]
```diff
@@ -43,13 +43,14 @@ def __init__(
             decoder=decoder,
         )
         if aioredis is None:
-            raise RuntimeError("Please install aioredis")
+            raise RuntimeError("Please install redis")
         # May have installed aioredis separately (without aiohttp-session[aioredis]).
-        if StrictVersion(aioredis.__version__).version < (2, 0):
-            raise RuntimeError("aioredis<2.0 is not supported")
+        lib_version = tuple(map(int, redis.__version__.split('.')[:2]))  # type: ignore
+        if lib_version < (4, 3):
```
Comment (Dreamsorcerer) [inline aiohttp_session/redis_storage.py:49]: Appears to already be a `redis.VERSION`, so there's no need to duplicate the tuple code here.

Comment (Dreamsorcerer) [reply aiohttp_session/redis_storage.py:49]: Also, adding to stubs: https://github.com/python/typeshed/pull/7838

File: aiohttp_session/redis_storage.py [hunk 3]
```diff
@@ -43,13 +43,14 @@ def __init__(
             decoder=decoder,
         )
         if aioredis is None:
-            raise RuntimeError("Please install aioredis")
+            raise RuntimeError("Please install redis")
         # May have installed aioredis separately (without aiohttp-session[aioredis]).
-        if StrictVersion(aioredis.__version__).version < (2, 0):
-            raise RuntimeError("aioredis<2.0 is not supported")
+        lib_version = tuple(map(int, VERSION[:2]))
```
Comment (Dreamsorcerer) [inline aiohttp_session/redis_storage.py:48]: This is either already an int, or this code will error. So, if we're confident they are ints, we should just do a `cast()`.

**Comment by Dreamsorcerer** [comment]
We'll need to fix up the remaining type errors, which might require some more changes to typeshed. I'll take a look at it eventually, if you don't get to it first.

**Comment by achimnol** [comment]
@Dreamsorcerer Could you review this again?

**Comment by amenezes** [comment]
@achimnol and @Dreamsorcerer,

Is there anything that I can make to help conclude this PR?

**Comment by Dreamsorcerer** [comment]
I'll get back to it in the coming weeks, but the tests are failing in current PRs, so we'll need to update and fix those problems as the initial priority (Think it was caused by missing the required check for dependabot, so it managed to merge some broken PRs).

File: aiohttp_session/redis_storage.py [hunk 4]
```diff
@@ -43,13 +43,14 @@ def __init__(
             decoder=decoder,
         )
         if aioredis is None:
-            raise RuntimeError("Please install aioredis")
+            raise RuntimeError("Please install redis")
         # May have installed aioredis separately (without aiohttp-session[aioredis]).
-        if StrictVersion(aioredis.__version__).version < (2, 0):
-            raise RuntimeError("aioredis<2.0 is not supported")
+        lib_version = VERSION[:2]
+        if lib_version < (4, 3):
```
Comment (Dreamsorcerer) [inline aiohttp_session/redis_storage.py:49]:
```suggestion
        if VERSION < (4, 3):
```

File: tests/test_redis_storage.py [hunk 5]
```diff
@@ -40,23 +42,24 @@ async def make_cookie(
...
     cookies = client.session.cookie_jar.filter_cookies(client.make_url("/"))
     key = cookies["AIOHTTP_SESSION"]
-    encoded = await redis.get("AIOHTTP_SESSION_" + key.value)
-    s = encoded.decode("utf-8")
-    value = json.loads(s)
+    value_bytes = await redis.get("AIOHTTP_SESSION_" + key.value)
+    if value_bytes is None:
+        return None
+    value = json.loads(value_bytes)
     return value
```
Comment (Dreamsorcerer) [inline tests/test_redis_storage.py:58]:
```suggestion
    return None if value_bytes is None else json.loads(value_bytes)
```

**Comment by Dreamsorcerer** [comment]
OK, someone has sorted out the tests already, had a moment to go through this PR again. Looking good, just a couple of tweaks I'd suggest.

**Pull request merged**